pdagent/sendevent.py

```diff
@@ -83,13 +84,12 @@ def tick(self):
             self.last_cleanup_time = int(time.time())
 
     def send_event(self, json_event_str, event_id):
-        # Note that Request here is from urllib2, not self._urllib2.
         request = Request(EVENTS_API_BASE)
         request.add_header("Content-type", "application/json")
-        request.add_data(json_event_str)
+        request.data = json_event_str.encode()
 
         try:
-            response = self._urllib2.urlopen(request,
+            response = self._http.urlopen(request,
```

Comment:
this call might not work? 🤔 i'd be curious how it does if it does :laughing:

Reply:
Looks like it's defined in [`sendevent.py`, line 63](https://github.com/PagerDuty/pdagent/blob/edd2d8a89e068240780fcece563a60aca857062b/pdagent/sendevent.py#L63).  It refers to the imported `pdagent.http` module, which _does_ contain a `urlopen()` function; in short, it looks like it should run just fine. 😄

Reply:
This caught me too, especially with the previous naming scheme. `urlopen` could use some rethinking but works for the time being.

Reply:
I don't think you end up using the `source_address` parameter in the new `urlopen` function, since the rest of the `kwargs` aren't used. Also mildly confused as to how it can use a `Request` object for the first param instead of a `url` as suggested by the function definition. 

Not sure if that's important enough though if it just works 😅

Reply:
Good catch on `source_address` -- looking into that now. I suspect testing there might not be comprehensive and will double-check against the previous implementation.

For `urlopen` taking a `Request`, either a URL or `Request` is supported though the signature and naming does make it confusing.

Reply:
Ah: and `source_address` looks like the only other parameter that we might send. (the optional params for cert and context don't appear to be used at all :laughing:)

Reply:
Updated to support `source_address` again.

It meant adding back a custom HTTPS handler to pass `source_address` down to the connection. At some point we should evaluate whether this is actually necessary.